### PR TITLE
support to build libspdm with vs2022

### PR DESCRIPTION
**Pull request by lbbxsxlz**
Signed-off-by: BinbinLee <lbbxsxlz@gmail.com>

fix the CI error.

### CODE REVIEW

**Comment by jyao1** [comment]
you can push the CI error to exiting PR, not need to submit a new one.

File: CMakeLists.txt [hunk 1]
```diff
@@ -506,7 +508,7 @@ elseif(CMAKE_SYSTEM_NAME MATCHES "Windows")
     else()
         # See https://gitlab.kitware.com/cmake/cmake/-/issues/18317 for following workaround
         STRING(REGEX REPLACE "/W3" "" CMAKE_C_FLAGS "${CMAKE_C_FLAGS}")
-        ADD_COMPILE_OPTIONS(/nologo /WX /W4 /Gs32768 /D UNICODE /Gy /EHs-c- /GR- /GF /Z7 /Gw /Oy- /D_CRT_SECURE_NO_WARNINGS /wd4063)
+        ADD_COMPILE_OPTIONS(/nologo /WX /W4 /Gs32768 /D UNICODE /Gy /EHs-c- /GR- /GF /Z7 /Gw /Oy- /D_CRT_SECURE_NO_WARNINGS /wd4063 /wd4819)
```
Comment (steven-bellock) [inline CMakeLists.txt:511]: Why do we need to disable this warning?

File: README.md [hunk 2]
```diff
@@ -81,11 +81,13 @@
 
 1) Compiler (Choose one)
 
-    a) [Visual Studio 2019](https://visualstudio.microsoft.com/vs/older-downloads/)
+    a) [Visual Studio 2022](https://visualstudio.microsoft.com/vs/older-downloads/)
```
Comment (steven-bellock) [inline README.md:84]: 2022 isn't listed in the URL.

File: CMakeLists.txt (hunk 1 above)
Comment (lbbxsxlz) [reply CMakeLists.txt:511]: > Why do we need to disable this warning?

In my PC(Windows 10 Home), there are Unicode Warning in files library/os_stub/include/industry_standard/link_type_ex.h and unit_test/cmockalib/cmocka/cmocka.h.

**Comment by jyao1** [comment]
Oh, I overlooked this.

> In my PC(Windows 10 Home), there are Unicode Warning in files library/os_stub/include/industry_standard/link_type_ex.h and unit_test/cmockalib/cmocka/cmocka.h.

I think we need replace the unicode with ascii.

**Comment by steven-bellock** [comment]
Yeah for libspdm-owned code we need to remove the unicode unless needed, and for non-libspdm code like cmocka we shouldn't be compiling with warnings enabled.

**Pull request merged**